gufe/protocols/protocol_settings.py

```diff
@@ -0,0 +1,381 @@
+from typing import Optional, Dict, Union, List
+
+from pydantic import BaseModel, validator
+from openff.units import unit
+
+
+class SystemSettings(BaseModel):
+    """Settings describing the simulation system settings.
+
+    Attributes
+    ----------
+    nonbonded_method : str
+        Which nonbonded electrostatic method to use, currently only PME
+        is supported.
+    nonbonded_cutoff : float * unit.nanometer
+        Cutoff value for short range interactions.
+        Default 1.0 * unit.nanometer.
+    constraints : str
+        Which bonds and angles should be constrained. Default None.
+    rigid_water : bool
+        Whether to apply rigid constraints to water molecules. Default True.
+    hydrogen_mass : float
+        How much mass to repartition to hydrogen. Default None, no
+        repartitioning will occur.
+    """
+    class Config:
+        arbitrary_types_allowed = True
```

Comment:
Just a note, this is required by pydantic to allow strange types.  In the future I think we could write validators for `openff.units` objects, then register these with pydantic somehow and not have to have this setting